Sources/MapboxMaps/Ornaments/OrnamentsManager.swift

```diff
@@ -39,271 +18,165 @@ public enum OrnamentVisibility: String, Equatable {
-        super.init()
-        self.view = view
-        self.ornamentConfig = ornamentConfig
+        // Scalebar View
+        scalebarView = MapboxScaleBarOrnamentView()
+        scalebarView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(scalebarView)
 
-        ensureTelemetryOptOutExists()
-    }
+        // Compass View
+        compassView = MapboxCompassOrnamentView(visibility: options.compassVisibility)
+        compassView.translatesAutoresizingMaskIntoConstraints = false
+        compassView.tapAction = { [weak view] in
+            view?.compassTapped()
+        }
+        view.addSubview(compassView)
 
-    internal func addOrnament(_ ornamentType: OrnamentType,
-                              at position: OrnamentPosition,
-                              visibility: OrnamentVisibility) {
-        let ornament = Ornament(view: nil, type: ornamentType, position: position, visibility: visibility)
-        ornamentConfig = ornamentConfig.with(ornament: ornament)
-    }
+        // Info Button
+        attributionButton = MapboxInfoButtonOrnament()
+        attributionButton.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(attributionButton)
 
-    internal func addOrnament(_ ornamentView: UIView, at position: OrnamentPosition) {
-        let ornament = Ornament(view: nil, type: .customOrnament(value: ornamentView), position: position)
-        ornamentConfig = ornamentConfig.with(ornament: ornament)
-    }
+        super.init()
 
-    internal func removeOrnament(_ ornamentView: UIView) {
-        ornamentConfig = OrnamentConfig(ornaments: ornaments.filter {
-            $0.view != ornamentView
-        })
-    }
+        updateOrnaments()
 
-    internal func removeOrnament(at position: OrnamentPosition) {
-        ornamentConfig = OrnamentConfig(ornaments: ornaments.filter {
-            $0.position != position
-        })
+        // Subscribe to updates for scalebar and compass
+        view.subscribeCameraChangeHandler { [scalebarView, compassView] (cameraOptions) in
+            if let zoom = cameraOptions.zoom,
+               let center = cameraOptions.center {
+                scalebarView.metersPerPoint = Projection.getMetersPerPixelAtLatitude(
+                    forLatitude: center.latitude,
+                    zoom: Double(zoom))
+            }
+            if let bearing = cameraOptions.bearing {
+                compassView.currentBearing = Double(bearing)
+            }
+        }
     }
 
-    internal func removeOrnament(with type: OrnamentType) {
-        ornamentConfig = OrnamentConfig(ornaments: ornaments.filter {
-            $0.type != type
-        })
-    }
+    internal func updateOrnaments() {
+        // Remove previously-added constraints
+        if constraints.count > 0 {
+            NSLayoutConstraint.deactivate(constraints)
+            constraints.removeAll()
+        }
```

Comment:
Does anything break if we remove the `if` check and just always call `deactivate()` and `removeAll()`?